### DISCUSSION TITLE
🤔 Favorite Git learning materials?

### DISCUSSION THREAD

**Discussion by MetzinAround** [community/community#114298] [GitHub Education] [answered]
### Hi everyone,

🤔 I was curious what you all use to learn git or to reinforce learning you already did. What are your favorite resources for helping yourself or others learn git concepts! 

I learned from friends teaching me directly, but was wondering what online resources y'all used! Let's get a thread going here for anyone learning!

**Comment by Tgentil** [comment]
One online resource that I found particularly helpful is [Learn Git Branching](https://learngitbranching.js.org/). This interactive site was a game-changer for me, especially when I was starting out in my first semester of college during the pandemic. It's excellent for visual learners as it really helps you understand what's happening with your Git commands.

You can also check out their GitHub project: [learnGitBranching on GitHub](https://github.com/pcottle/learnGitBranching). It's an interactive way to get a grasp of Git concepts, and I think it could be really beneficial for both beginners and those looking to reinforce their knowledge.

Hope this helps and adds to your learning journey with Git!

**Reply by MetzinAround** [reply]
Love me some visual learning, esPECIALLY for branching. Thanks!

**Comment by mwanyumba7** [comment]
If you enjoy using CLI you can checkout [esgit — easy-git](https://github.com/dillionmegida/esgit) - CLI package with simplified aliases for git commands. By [Dillion Megida](https://twitter.com/iamdillion)

Can be a great way for one to practice on the go :-)

**Comment by mishmanners** [comment, marked as answer]
I have a whole list of free learning content online: https://github.com/mishmanners/Free-Learning-Content

My favourite git ones are: GitHub Skills, FreeCodeCamp, and some of the things on DEV (lots of articles there on git).

**Comment by devashish2024** [comment]
This is the official and best, simple and short cheatsheet I'd recommend.

https://education.github.com/git-cheat-sheet-education.pdf

**Comment by priyanujgogoi-28** [comment]
Firstly, please don't refer to any quick tutorial/guide site for learning something new. They are mostly outdated, aren't in-depth, one/two-paragraph explanations, not heavily invested, unverified e-resources etc. However, you can try checking them out later after having a base understanding of that *something*. This way, you'll know that you aren't learning it wrongly.

Secondly, always prefer the official doc site as reference at first. It contains most recent and updated materials. You don't need to check out the whole reference at once though. Just keep progressing after you've understood a topic and also, have explored that as practically too (optional but preferred).

So for Git, I recommend to check out: https://git-scm.com/doc

PS: I learnt Git just by checking the official doc reference and testing each checked-commands practically through trial-error in the terminal.

**Reply by MetzinAround** [reply]
That's some great feedback! Don't learn quick, learn deep at first. Then you can use quick learning later to boost smaller concepts within the larger context!
